community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/ReadNodeRecordsStep.java

```diff
@@ -52,9 +52,8 @@ protected Object nextBatchOrNull( long ticket, int batchSize )
         NodeRecord[] batch = new NodeRecord[size];
         for ( int i = 0; i < size; i++ )
         {
-            // Passing in null creates a new record if it's in use. Returning null is fine since our processor,
-            // i.e. consumer of these batches, will filter those out.
-            batch[i] = nodeStore.loadRecord( id++, null );
+            // We don't want null in batch[i], a not used record is what we want
+            nodeStore.loadRecord( id, batch[i] = new NodeRecord( id++ ) );
```

Comment:
I think it was due to one scenario needing to call `NodeStore#updateRecord` with a NodeRecord that wasn't in use, as opposed to _not_ calling `updateRecord`. And so moved away from `null` in favor `!inUse` records as a whole.